src/Components/Main.js

```diff
@@ -1,51 +1,41 @@
-import React, {Component} from 'react';
-import { mockedMovies } from './Consts';
-import MovieCard from "./MovieCard";
-import {MovieModal} from "./MovieModal";
-
-class Main extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isEditMovieModalShown: false,
-            isDeleteModalShown:false,
-        }
-    }
+import React from 'react';
+import PropTypes from 'prop-types';
 
-    toggleModal = (name) => {
-        this.setState(prevState => ({
-            [name]: !prevState[name]
-        }))
-    }
+import { mockedMovies } from './Consts';
+import { MovieCard } from "./MovieCard";
+import { MovieModal } from "./MovieModal";
 
-    handleToggleEditMovieModal = () => {
-        this.toggleModal('isEditMovieModalShown')
-    }
+export const Main = ({ useToggle, setMovieInfo, showMovieInfo }) => {
+    const [isEditMovieModalShown, toggleEditMovieModal] = useToggle(false);
+    const [isDeleteModalShown, toggleDeleteMovieModal] = useToggle(false);
 
-    handleToggleDeleteModal = () => {
-        this.toggleModal('isDeleteModalShown')
-    }
+    return (
+        <>
+            <div className="movie-list">
+                {mockedMovies.map(movie =>
+                    <MovieCard
+                        key={movie.id}
+                        movie={movie}
+                        toggleEditMovieModal={toggleEditMovieModal}
+                        toggleDeleteMovieModal={toggleDeleteMovieModal}
+                        useToggle={useToggle}
+                        setMovieInfo={setMovieInfo}
+                        showMovieInfo={showMovieInfo}
```

Comment:
do you need to pass it to `MovieCard` component?

Reply:
no) fixed